TF_Settings_Web/src/Components/Pages/CameraPage.tsx

```diff
@@ -1,18 +1,240 @@
-import { CSSProperties } from "react";
+import { TextEntry } from "../Controls/TextEntry";
+
+import '../../Styles/CameraPage.css';
 
 import { Page } from "./Page";
+import { PhysicalConfig } from "../../TouchFree/Configuration/ConfigurationTypes";
+import { ConfigurationManager } from "../../TouchFree/Configuration/ConfigurationManager";
+import { ConfigState, WebSocketResponse } from "../../TouchFree/Connection/TouchFreeServiceTypes";
+import { ConnectionManager } from "../../TouchFree/Connection/ConnectionManager";
+
+interface physicalState {
+    screenHeight: number;
+    cameraHeight: number;
+    cameraLeftToRight: number;
+    screenTilt: number;
+    cameraRotation: number;
+    cameraDistanceFromScreen: number;
+    physicalConfig: PhysicalConfig;
+    selectedView: string;
+};
+
+export class CameraPage extends Page<{}, physicalState> {
+    constructor(props: {}) {
+        super(props);
+
+        let state: physicalState = {
+            screenHeight: 0,
+            cameraHeight: 0,
+            cameraLeftToRight: 0,
+            screenTilt: 0,
+            cameraRotation: 0,
+            cameraDistanceFromScreen: 0,
+            physicalConfig: {LeapPositionRelativeToScreenBottomM:{X:0,Y:0,Z:0}, LeapRotationD:{X:0,Y:0,Z:0}, ScreenHeightM:0, ScreenRotationD:0},
+            selectedView: "screenHeight"
+        };
+
+        this.state = state;
+    }
+
+    componentDidMount(): void {
+        if (ConnectionManager.serviceConnection()?.webSocket?.readyState === WebSocket.OPEN) {
+            ConfigurationManager.RequestConfigFileState(this.setStateFromFile.bind(this));
+        } else {
+            ConnectionManager.instance.addEventListener('OnConnected', () => {
```

Comment:
ConnectionManager has a custom `AddConnectionListener` function which could replace the logic of this function (if already connected, the callback is immediately executed, if not, it's added to a list to be invoked once a connection is made)